src/plugins/controls/common/types.ts

```diff
@@ -30,5 +30,7 @@ export type ControlInput = EmbeddableInput & {
 
 export type DataControlInput = ControlInput & {
   fieldName: string;
+  parentFieldName?: string;
```

Comment:
This is a great way to do it!

Just as a note to ourselves for the future, we might eventually want to replace this with an array of `linkedFields`, so that we can support field types with many sub fields like the [search-as-you-type](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-as-you-type.html) mapping type.